front/lib/api/assistant/configuration.ts

```diff
@@ -1030,3 +1030,148 @@ async function _createAgentDataSourcesConfigData(
+            providerId: agentConfiguration.generationConfiguration.providerId,
+            modelId: agentConfiguration.generationConfiguration.modelId,
+            temperature: agentConfiguration.generationConfiguration.temperature,
+          },
+          { transaction: t }
+        );
+      }
+
+      let newRetrievalConfiguration: AgentRetrievalConfiguration | null = null;
+      if (agentConfiguration.retrievalConfigurationId) {
+        newRetrievalConfiguration = await AgentRetrievalConfiguration.create(
+          {
+            sId: generateModelSId(),
+            query: agentConfiguration.retrievalConfiguration.query,
+            queryTemplate:
+              agentConfiguration.retrievalConfiguration.queryTemplate,
+            relativeTimeFrame:
+              agentConfiguration.retrievalConfiguration.relativeTimeFrame,
+            relativeTimeFrameDuration:
+              agentConfiguration.retrievalConfiguration
+                .relativeTimeFrameDuration,
+            relativeTimeFrameUnit:
+              agentConfiguration.retrievalConfiguration.relativeTimeFrameUnit,
+            topK: agentConfiguration.retrievalConfiguration.topK,
+            topKMode: agentConfiguration.retrievalConfiguration.topKMode,
+          },
+          { transaction: t }
+        );
+      }
+
+      let newAgentDustAppRunConfiguration: AgentDustAppRunConfiguration | null =
+        null;
+      if (agentConfiguration.dustAppRunConfigurationId) {
+        newAgentDustAppRunConfiguration =
+          await AgentDustAppRunConfiguration.create(
+            {
+              sId: generateModelSId(),
+              appWorkspaceId:
+                agentConfiguration.dustAppRunConfiguration.appWorkspaceId,
+              appId: agentConfiguration.dustAppRunConfiguration.appId,
+            },
+            { transaction: t }
+          );
+      }
+      let newAgentDatabaseQueryConfiguration: AgentDatabaseQueryConfiguration | null =
+        null;
+      if (agentConfiguration.databaseQueryConfigurationId) {
+        newAgentDatabaseQueryConfiguration =
+          await AgentDatabaseQueryConfiguration.create(
+            {
+              sId: generateModelSId(),
+              dataSourceWorkspaceId:
+                agentConfiguration.databaseQueryConfiguration
+                  .dataSourceWorkspaceId,
+              dataSourceId:
+                agentConfiguration.databaseQueryConfiguration.dataSourceId,
+              databaseId:
+                agentConfiguration.databaseQueryConfiguration.databaseId,
+            },
+            { transaction: t }
+          );
+      }
+
+      const newAgentSid = generateModelSId();
+      const newAgentConfiguration = await AgentConfiguration.create(
+        {
+          sId: newAgentSid,
+          version: agentConfiguration.version,
+          status: agentConfiguration.status,
+          scope: "workspace", // need to become private in the furue
```

Comment:
This is for testing purposes and you intend to switch in a following PR, correct?

Reply:
Putting it to private would not make it show up at all, so I assumed the private listing was not taken care of yet. Am I wrong?

Reply:
It should definitely be `private` but we don't show them yet indeed in the "My assistants" list. But it should show up in your assistant selector in converstaions.